gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -619,11 +619,28 @@ protected static DiffPathSet fullPathDiff(HiveLocationDescriptor sourceLocation,
       if (shouldCopy) {
         builder.copyFile(sourcePath);
       } else {
-        // if not copying, we want to keep the file in the target
+        // If not copying, we want to keep the file in the target
         // at the end of this loop, all files in targetExistingPaths will be marked for deletion, so remove this file
         targetExistingPaths.remove(newPath);
         desiredTargetExistingPaths.remove(newPath);
       }
+
+      // If target already has the path and it should be copied,
+      // delete the existing target path, even if deleteMethod is NO_DELETE
+      if (targetExistingPaths.containsKey(newPath) && shouldCopy) {
+        builder.deleteFile(newPath);
+        targetExistingPaths.remove(newPath);
+        desiredTargetExistingPaths.remove(newPath);
+      }
+    }
+
+    // If deleteMethod is NO_DELETE, clear targetExistingPaths,
+    // so nothing goes to pathsToDelete
+    if (helper.deleteMethod == DeregisterFileDeleteMethod.NO_DELETE) {
```

Comment:
This logic is incorrect. NO_DELETE should only apply when the location of the table is changing. As it is right now, distcp-ng will not be able to sync directories when NO_DELETE is set.